src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -479,15 +480,12 @@ trait Implicits { self: Typer =>
       formal.argTypes match {
         case arg :: Nil =>
           val tp = fullyDefinedType(arg, "ClassTag argument", pos)
-          tp.underlyingClassRef(refinementOK = false) match {
-            case tref: TypeRef =>
-              return ref(defn.ClassTagModule)
-                .select(nme.apply)
-                .appliedToType(tp)
-                .appliedTo(clsOf(tref))
-                .withPos(pos)
-            case _ =>
-          }
+          if (hasStableErasure(tp))
+            return ref(defn.ClassTagModule)
+              .select(nme.apply)
+              .appliedToType(tp)
+              .appliedTo(clsOf(erasure(tp)))
```

Comment:
It covers about 30 ops, maybe half of them construct syntax trees. See `ast.tpd#TreeOps` for where it  is defined.